### ISSUE TITLE
Option 'linear' defaults to True in ClassBlock

### ISSUE THREAD

**Issue by CynicalHeart** [layumi/Person_reID_baseline_pytorch#302]
> In ClassBlock class, the option'linear' defaults to True will add a linear layer that loses accuracy.

```python
def __init__(self, input_dim, class_num, droprate, relu=False, bnorm=True, num_bottleneck=512, linear=True, return_f = False):
        super(ClassBlock, self).__init__()
        self.return_f = return_f
        add_block = []
        if linear:
            add_block += [nn.Linear(input_dim, num_bottleneck)]  # Resnet feature dim form 2048 to 512
        ...
```

In multiple encapsulated models, there is no parametric customization of this option（linear=True）. 

When i use `--ibn`,  i found that the feature dimension has become 512.

I did a comparative experiment，the network structure is r50-ibn, the loss function is circle loss, and other tricks remain the same.
- 512：Rank1 - 92.2% mAP: 80.2%
- 2048: Rank1 - 92.7% mAP: 82.15%

Perhaps it is for query or calculation efficiency improvement, but this parameter should be used as a option, as one of the considerations for comparison.

**Comment by layumi** [comment]
Thanks a lot @CynicalHeart 
The linear is to control the feature dimension. 
We will test this option.

**Comment by layumi** [comment]
Hi @kagawa123  Mu

Could you try to modify the code? 
You need to modify the test to support 2048-dim as well. Please use the new config to make it right.

**Comment by kagawa123** [comment]
> 

Sure, I will modify it!

**Comment by CynicalHeart** [comment]
Already achieved！

**Comment by yg76cv** [comment]
Sorry， I would like to ask if this code ‘num_bottleneck = linear’  was added this time. An error will occur when running in the model.
![image](https://user-images.githubusercontent.com/90538536/148347280-4a53f4d9-33b2-4771-b716-91a0a501c970.png)
and this is my code
![image](https://user-images.githubusercontent.com/90538536/148347391-3bbc34a2-fba0-4ffc-ad69-25c88109896a.png)
